### PR TITLE
PUBLIC-2591: Dockerfile update

### PR DESCRIPTION
**Pull request by ialarmedalien**
- Update dockerfile
- Remove unneeded files (the Fasttree source must have been downloaded locally at some point and added to this repo; it isn't needed).
- Switch to generating compile_report.json prior to image build

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -1,12 +1,10 @@
 name: "CodeQL"
 
 on:
-    push:
```
Comment (ialarmedalien) [inline .github/workflows/codeql.yml:4]: doesn't need to be run on `push` as well

File: Dockerfile [hunk 2]
```diff
@@ -1,44 +1,28 @@
...
-
-
-# Install ETE3
-RUN apt-get -y install xvfb
-RUN pip install --upgrade pip
-# Note: You must use PyQt5==5.11.3 on debian
-RUN pip install ete3==3.1.2 PyQt5==5.11.3 numpy==1.23.1 pytest coverage pytest-cov
-
-# -----------------------------------------
+RUN apt-get update \
```
Comment (ialarmedalien) [inline Dockerfile:6]: deduplicating and compressing the commands

File: Dockerfile [hunk 3]
```diff
@@ -1,44 +1,28 @@
...
-# Install FastTree
-#
-RUN mkdir -p /kb/module/FastTree/bin
-WORKDIR /kb/module/FastTree/bin
-RUN curl -o FastTree http://www.microbesonline.org/fasttree/FastTree && \
-    chmod 555 FastTree
 
+RUN mkdir -p /kb/module/work && \
+    chmod -R a+rw /kb/module && \
+    mv /kb/module/compile_report.json /kb/module/work/compile_report.json && \
```
Comment (ialarmedalien) [inline Dockerfile]: this copies the compile report to where the catalog expects it to be

File: scripts/entrypoint.sh [hunk 4]
```diff
@@ -22,7 +22,7 @@ elif [ "${1}" = "bash" ] ; then
   bash
 elif [ "${1}" = "report" ] ; then
   export KB_SDK_COMPILE_REPORT_FILE=./work/compile_report.json
-  make compile
+  echo The compile report is available at /kb/module/work/compile_report.json
```
Comment (ialarmedalien) [inline scripts/entrypoint.sh:25]: the compile_report is already part of the repo so there is no need to create it (and all the other files) again.

Comment (bio-boris) [reply scripts/entrypoint.sh:25]: How does a dev know whether or not to recompile it and add it into the repo?

Comment (ialarmedalien) [reply scripts/entrypoint.sh:25]: See the README -- I added these lines:

> ## API changes
> 
>  If you make changes to the spec (and thus the API) of the kb_fasttree app, you will need to recompile the app to generate updated versions of `kb_fasttreeImpl.py`, `kb_fasttreeServer.py`, etc., and the compilation report, `compile_report.json`. These updated files must be committed to the GitHub repo.
> 
>  Use the command `make compile` (or simply `make`) to regenerate these files.

Comment (bio-boris) [reply scripts/entrypoint.sh:25]: Maybe we should add an automation for this that changes it, or possibly even does the compilation for you?

```
name: Check .spec and .compile_report.json Changes

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  check-files-change:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Check for .spec file changes
        id: check_spec_files
        run: |
          # Check if any .spec files were changed
          if git diff --name-only HEAD~1 HEAD | grep -q '\.spec$'; then
            echo ".spec file was changed"
            echo "spec_changed=true" >> $GITHUB_ENV
          else
            echo "No .spec files were changed"
            echo "spec_changed=false" >> $GITHUB_ENV

      - name: Check for .compile_report.json change
        id: check_compile_report
        run: |
          # Check if .compile_report.json was changed
          if git diff --name-only HEAD~1 HEAD | grep -q '.compile_report.json'; then
            echo ".compile_report.json was changed"
            echo "report_changed=true" >> $GITHUB_ENV
          else
            echo ".compile_report.json was not changed"
            echo "report_changed=false" >> $GITHUB_ENV

      - name: Validate changes
        if: env.spec_changed == 'true'
        run: |
          if [ "$report_changed" != "true" ]; then
            echo "Error: .spec files were changed but .compile_report.json was not changed."
            exit 1
          else
            echo "Validation passed: Both .spec files and .compile_report.json were changed."
          fi

      - name: Proceed if validation passed
        if: env.spec_changed == 'false' || env.report_changed == 'true'
        run: |
          echo "Proceeding with the next steps"
          # Add your additional actions here
```

Comment (bio-boris) [reply scripts/entrypoint.sh:25]: I think it would be nice to add "Spec File" to that section heading!

Comment (ialarmedalien) [reply scripts/entrypoint.sh:25]: I think this check would be really helpful as a precommit hook -- it means that there's no "magic" happening when the files hit GitHub and the dev should be rerun any tests, etc., before they redo the commit and then push the files to GitHub.

**Pull request merged**